src/storage/sqlite3_storage_engine.cpp

```diff
@@ -777,9 +789,23 @@ std::vector<int> Sqlite3StorageEngine::QueryIdsBySearch(
  */
 bool Sqlite3StorageEngine::Store(ITelemetry data)
 {
-    if (!ready)
-        return false;
+    if (!ready) return false;
 
+    // Update the syscallHitMap map to keep total running syscalls and durations. We store it here
+    // in a shared map to avoid the cost of keeping an additional table. The map is sorted by duration
+    // only when user requests it through the 'Stats' capability.
```

Comment:
Can we not compute this information using a query on the db?

Something like:
```sql
select syscall, sum(duration), count(*) as hits
from ebpf
group by syscall
```

I'm wary of polluting the insertion code path with too much extra work.

Reply:
This overhead at insertion time is not new. I moved the logic out of ebpf tracer and into the database engine so that the stats can be included in an export.

As for the suggestion to query the DB it could be done however the overhead of calculating this at load time of a large export (1M+ records) would not be ideal for post mortem analysis.

Reply:
I wouldn't expect a calculation like that to take too long even with millions of events, but I could be wrong. We could add an index on the table on `syscall` which would speed it up. We should definitely cache the result though, since it makes no sense to recalculate it every time the detail window is opened. Even if the overhead existed before these changes, we should try to remove anything in the insertion path that is extra. It might be outside the scope of these changes though.

Reply:
The sqlite overhead is very large when you have 1M+ records. For a preview of it let procmon run for awhile, collect a large amount of records, and then start page scrolling rapidly. You'll immediately notice the latency due to the small queries of trying to grab the records for the next page.

I would agree that optimization on the insertion path is out of scope of these changes. These changes are aimed at fixing the missing stats when you load an exported trace.

Reply:
I'm fine not changing this in this PR, but I would like to go through and see if we can squeeze out as much perf out of sqlite as possible. That can be later work though.